packages/toolkit/src/query/tests/useMutation-fixedCacheKey.test.tsx

```diff
@@ -354,4 +357,34 @@ describe('fixedCacheKey', () => {
     expect(getByTestId(c1, 'status').textContent).toBe('fulfilled')
     expect(getByTestId(c1, 'data').textContent).toBe('this should be visible')
   })
+
+  test('using fixedCacheKey should create a new cache entry', async () => {
+    api.injectEndpoints({
+      overrideExisting: true,
+      endpoints: (build) => ({
+        send: build.mutation<string, string | Promise<string>>({
+          query: (arg) => arg,
+          onCacheEntryAdded(arg, {}) {
+            onNewCacheEntry(arg)
+          },
+        }),
+      }),
+    })
```

Comment:
```suggestion
    api.enhanceEndpoints({
      endpoints: {
        send: {
          onCacheEntryEnded: (arg) => onNewCacheEntry(arg)
        }
      }
    })
```

Reply:
sorry, my suggestion had a typo - edited

the problem with what you have now is that your mock isn't being reset between tests

<img width="624" alt="image" src="https://github.com/reduxjs/redux-toolkit/assets/18308300/d3a37de5-df94-4b8b-b369-f296120bceb7">

Reply:
All good, I thought the ended was a typo.

Seeing as you're active, I'll ask what the preference here is so I can know for future as I am not an expert on unit testing practices really. Is it preferred that I create the endpoint override within the scope of that single test, or have the functionality added to the initial declaration and add and `afterEach () => {clear mock function}`

Reply:
i think you should create the mock and call enhanceEndpoints within the single test, and at the end of the test block do 
```ts
    api.enhanceEndpoints({
      endpoints: {
        send: {
          onCacheEntryAdded: undefined,
        },
      },
    })
```
to make sure it doesn't continue onto other tests

Reply:
That's why you get paid the big bucks, Ben. Recent commit should reflect this change.